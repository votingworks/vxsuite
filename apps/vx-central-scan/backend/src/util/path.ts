import { isAbsolute, join } from 'path';

/**
 * Joins path parts by using the last absolute path, if any, and joining
 * all the following parts, normalizing the result.
 */
export function normalizeAndJoin(path: string, ...parts: string[]): string {
  let result = path;

  for (const part of parts) {
    if (isAbsolute(part)) {
      result = part;
    } else {
      result = join(result, part);
    }
  }

  return result;
}
