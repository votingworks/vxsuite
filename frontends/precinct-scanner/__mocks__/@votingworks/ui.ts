// eslint-disable-next-line import/no-extraneous-dependencies, import/no-import-module-exports
import {
  fakePrintElementWhenReady,
  fakePrintElement,
} from '@votingworks/test-utils';

const ui = jest.requireActual('@votingworks/ui');

module.exports = {
  ...ui,
  printElement: fakePrintElement,
  printElementWhenReady: fakePrintElementWhenReady,
};
