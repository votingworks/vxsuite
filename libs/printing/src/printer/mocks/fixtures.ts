import {
  IppMarkerInfo,
  PrinterConfig,
  PrinterRichStatus,
  PrinterStatus,
} from '@votingworks/types';

export const MOCK_MARKER_INFO: IppMarkerInfo = {
  color: '#000000',
  highLevel: 100,
  level: 100,
  lowLevel: 2,
  name: 'black cartridge',
  type: 'toner-cartridge',
};

export const MOCK_PRINTER_RICH_STATUS: PrinterRichStatus = {
  state: 'idle',
  stateReasons: [],
  markerInfos: [MOCK_MARKER_INFO],
};

export function getMockConnectedPrinterStatus(
  config: PrinterConfig
): PrinterStatus {
  if (config.supportsIpp) {
    return {
      connected: true,
      config,
      richStatus: MOCK_PRINTER_RICH_STATUS,
    };
  }

  return {
    connected: true,
    config,
  };
}
