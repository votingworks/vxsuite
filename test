#!/usr/bin/env bun

import { readFile } from 'fs/promises';
import { join } from 'path';

const { $ } = Bun;
export {};

const workspacePackageJson = await readFile(
  join(__dirname, 'package.json'),
  'utf8'
);
const { workspaces } = JSON.parse(workspacePackageJson);
const WIDTH = 100;

const results: { pkgPath: string; status: 'success' | 'failure' }[] = [];
let totalPassed = 0;
let totalFailed = 0;

for (const pkgPath of workspaces) {
  const tests = box(
    (async function* () {
      yield `→ bun run test --cwd ${pkgPath}`;
      const bunTest = $`bun run test --cwd ${pkgPath}`.quiet();

      const bunTestResult = await bunTest;

      yield* bunTestResult.stderr.toString('utf-8').split('\n');
      yield `${bunTestResult.exitCode === 0 ? '✅' : '❌'} ${pkgPath}`;

      if (bunTestResult.exitCode === 0) {
        results.push({ pkgPath, status: 'success' });
      } else {
        results.push({ pkgPath, status: 'failure' });
      }
    })()
  );

  let passed = 0;
  let failed = 0;

  for await (const line of tests) {
    const passedMatch = line.match(/(\d+) pass/);
    const failedMatch = line.match(/(\d+) fail/);

    if (passedMatch) {
      passed = parseInt(passedMatch[1], 10);
    }

    if (failedMatch) {
      failed = parseInt(failedMatch[1], 10);
    }

    console.log(line);
  }

  totalPassed += passed;
  totalFailed += failed;
}

console.log(
  `\n${totalPassed + totalFailed} tests: ${totalPassed} passed, ${totalFailed} failed`
);

for (const { pkgPath, status } of results) {
  console.log(`${status === 'success' ? '✅' : '❌'} ${pkgPath}`);
}

async function* box(lines: AsyncIterable<string>): AsyncIterable<string> {
  yield* lines;
}
