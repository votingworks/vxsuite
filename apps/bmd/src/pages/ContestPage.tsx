import React, { useContext, useEffect, useState } from 'react'
import { RouteComponentProps } from 'react-router-dom'
import { CandidateVote, OptionalYesNoVote } from '@votingworks/types'

import ordinal from '../utils/ordinal'

import BallotContext from '../contexts/ballotContext'

import CandidateContest from '../components/CandidateContest'
import ElectionInfo from '../components/ElectionInfo'
import LinkButton from '../components/LinkButton'
import Prose from '../components/Prose'
import Screen from '../components/Screen'
import Sidebar from '../components/Sidebar'
import Text from '../components/Text'
import YesNoContest from '../components/YesNoContest'
import SettingsTextSize from '../components/SettingsTextSize'
import TextIcon from '../components/TextIcon'
import MsEitherNeitherContest from '../components/MsEitherNeitherContest'

interface ContestParams {
  contestNumber: string
}

const ContestPage: React.FC<RouteComponentProps<ContestParams>> = (props) => {
  const isReviewMode = window.location.hash === '#review'
  const { contestNumber } = props.match.params
  const {
    ballotStyleId,
    contests,
    electionDefinition,
    precinctId,
    setUserSettings,
    updateVote,
    userSettings,
    votes,
    machineConfig,
  } = useContext(BallotContext)
  const { election } = electionDefinition
  const currentContestIndex = parseInt(contestNumber, 10)
  const contest = contests[currentContestIndex]

  const vote = votes[contest.id]

  const [isVoteComplete, setIsVoteComplete] = useState(false)

  const prevContestIndex = currentContestIndex - 1
  const prevContest = contests[prevContestIndex]

  const nextContestIndex = currentContestIndex + 1
  const nextContest = contests[nextContestIndex]

  useEffect(() => {
    const calculateIsVoteComplete = () => {
      /* istanbul ignore else */
      if (contest.type === 'yesno') {
        setIsVoteComplete(!!vote)
      } else if (contest.type === 'candidate') {
        setIsVoteComplete(
          contest.seats === ((vote as CandidateVote) ?? []).length
        )
      } else if (contest.type === 'ms-either-neither') {
        setIsVoteComplete(
          votes[contest.pickOneContestId]?.length === 1 ||
            votes[contest.eitherNeitherContestId]?.[0] === 'no'
        )
      }
    }
    calculateIsVoteComplete()
  }, [contest, vote, votes])

  return (
    <Screen>
      {contest.type === 'candidate' && (
        <CandidateContest
          aria-live="assertive"
          key={contest.id}
          contest={contest}
          parties={election.parties}
          vote={(vote ?? []) as CandidateVote}
          updateVote={updateVote}
        />
      )}
      {contest.type === 'yesno' && (
        <YesNoContest
          key={contest.id}
          contest={contest}
          vote={vote as OptionalYesNoVote}
          updateVote={updateVote}
        />
      )}
      {contest.type === 'ms-either-neither' && (
        <MsEitherNeitherContest
          key={contest.id}
          contest={contest}
          eitherNeitherContestVote={
            votes[contest.eitherNeitherContestId] as OptionalYesNoVote
          }
          pickOneContestVote={
            votes[contest.pickOneContestId] as OptionalYesNoVote
          }
          updateVote={updateVote}
        />
      )}
      <Sidebar
        footer={
          <React.Fragment>
            <SettingsTextSize
              userSettings={userSettings}
              setUserSettings={setUserSettings}
            />
            <ElectionInfo
              electionDefinition={electionDefinition}
              ballotStyleId={ballotStyleId}
              precinctId={precinctId}
              machineConfig={machineConfig}
              horizontal
            />
          </React.Fragment>
        }
      >
        <Prose>
          <Text center>
            This is the <strong>{ordinal(currentContestIndex + 1)}</strong> of{' '}
            {contests.length} contests.
          </Text>
          {isReviewMode ? (
            <React.Fragment>
              <p>
                <LinkButton
                  big
                  primary={isVoteComplete}
                  to={`/review#contest-${contest.id}`}
                  id="next"
                >
                  <TextIcon arrowRight white={isVoteComplete}>
                    Review
                  </TextIcon>
                </LinkButton>
              </p>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <p>
                <LinkButton
                  big
                  id="next"
                  primary={isVoteComplete}
                  to={nextContest ? `/contests/${nextContestIndex}` : '/review'}
                >
                  <TextIcon arrowRight white={isVoteComplete}>
                    Next
                  </TextIcon>
                </LinkButton>
              </p>
              <p>
                <LinkButton
                  small
                  id="previous"
                  to={prevContest ? `/contests/${prevContestIndex}` : '/'}
                >
                  <TextIcon arrowLeft>Back</TextIcon>
                </LinkButton>
              </p>
            </React.Fragment>
          )}
        </Prose>
      </Sidebar>
    </Screen>
  )
}

export default ContestPage
