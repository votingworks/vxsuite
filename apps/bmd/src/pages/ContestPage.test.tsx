import { join } from 'path'
import React from 'react'
import { Route } from 'react-router-dom'
import { loadElectionDefinition } from '@votingworks/fixtures'

import { render } from '../../test/testUtils'
import ContestPage from './ContestPage'

const electionSampleDefinition = loadElectionDefinition(
  join(__dirname, '../data/electionSample.json')
)

const firstContestTitle = electionSampleDefinition.election.contests[0].title

it('Renders ContestPage', () => {
  const { container, getByText } = render(
    <Route path="/contests/:contestNumber" component={ContestPage} />,
    {
      route: '/contests/0',
    }
  )
  getByText(firstContestTitle)
  expect(container).toMatchSnapshot()
})
