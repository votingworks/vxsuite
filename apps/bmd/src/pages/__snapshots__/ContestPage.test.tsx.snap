// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ContestPage 1`] = `
.c14 {
  border: none;
  border-radius: 0.25rem;
  box-shadow: 0 0 0 0 rgba(71,167,75,1);
  box-sizing: border-box;
  background: rgb(211,211,211);
  cursor: pointer;
  padding: 1rem 1.75rem;
  line-height: 1.25;
  color: black;
  font-size: 1.25rem;
  touch-action: manipulation;
}

.c16 {
  border: none;
  border-radius: 0.25rem;
  box-shadow: 0 0 0 0 rgba(71,167,75,1);
  box-sizing: border-box;
  background: rgb(211,211,211);
  cursor: pointer;
  padding: 0.35rem 0.5rem;
  line-height: 1.25;
  color: black;
  touch-action: manipulation;
}

.c24 {
  border: none;
  border-radius: 0.25rem;
  box-shadow: 0 0 0 0 rgba(71,167,75,1);
  box-sizing: border-box;
  background: rgb(71,167,75);
  cursor: pointer;
  padding: 0.35rem 0.5rem;
  line-height: 1.25;
  color: #FFFFFF;
  touch-action: manipulation;
}

.c23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  white-space: nowrap;
}

.c23 > button {
  box-shadow: inset 1px 0 0 rgb(190,190,190);
}

.c23 > button:first-child {
  box-shadow: none;
}

.c23 > button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c23 > button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c23 > button:disabled {
  background: #028099;
  color: #ffffff;
}

.c9 {
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #FFFFFF;
  cursor: pointer;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
  text-align: left;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

.c9::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: rgb(211,211,211);
  border-radius: 0.125rem 0 0 0.125rem;
  background: #ffffff;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: '';
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.c3 {
  max-width: 66ch;
  line-height: 1.2;
}

.c3 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c3 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c3 h3 {
  font-size: 1.17rem;
}

.c3 h4 {
  font-size: 1rem;
}

.c3 h5 {
  font-size: 0.9rem;
}

.c3 h3,
.c3 h4,
.c3 h5,
.c3 p,
.c3 ol,
.c3 ul {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p,
.c3 h1 + ol,
.c3 h2 + ol,
.c3 h1 + ul,
.c3 h2 + ul {
  margin-top: -0.75rem;
}

.c3 h3 + p,
.c3 h4 + p,
.c3 h5 + p,
.c3 h3 + ol,
.c3 h4 + ol,
.c3 h5 + ol,
.c3 h3 + ul,
.c3 h4 + ul,
.c3 h5 + ul {
  margin-top: -1rem;
}

.c3 > :first-child {
  margin-top: 0;
}

.c3 > :last-child {
  margin-bottom: 0;
}

.c3 dl {
  margin: 1rem 0;
}

.c28 {
  max-width: 66ch;
  line-height: 1.2;
}

.c28 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c28 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c28 h3 {
  font-size: 1.17rem;
}

.c28 h4 {
  font-size: 1rem;
}

.c28 h5 {
  font-size: 0.9rem;
}

.c28 h3,
.c28 h4,
.c28 h5,
.c28 p,
.c28 ol,
.c28 ul {
  margin-top: 0;
  margin-bottom: 0;
}

.c28 h1 + h2 {
  margin-top: -0.75rem;
}

.c28 h1 + p,
.c28 h2 + p,
.c28 h1 + ol,
.c28 h2 + ol,
.c28 h1 + ul,
.c28 h2 + ul {
  margin-top: -0.75rem;
}

.c28 h3 + p,
.c28 h4 + p,
.c28 h5 + p,
.c28 h3 + ol,
.c28 h4 + ol,
.c28 h5 + ol,
.c28 h3 + ul,
.c28 h4 + ul,
.c28 h5 + ul {
  margin-top: 0;
}

.c28 > :first-child {
  margin-top: 0;
}

.c28 > :last-child {
  margin-bottom: 0;
}

.c28 dl {
  margin: 1rem 0;
}

.c10 {
  word-break: break-word;
}

.c13 {
  text-align: center;
}

.c29 {
  font-size: 0.8rem;
}

.c31 {
  font-size: 0.8rem;
  font-weight: 300;
}

.c30 {
  white-space: nowrap;
}

.c2 {
  margin: 0 auto;
  width: 100%;
  padding: 1rem 2rem 0.5rem;
  padding-left: 6rem;
}

.c4 {
  text-transform: uppercase;
  font-size: 0.85rem;
  font-weight: 600;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  overflow: auto;
}

.c5::before,
.c5::after {
  position: absolute;
  z-index: 1;
  width: 100%;
  height: 0.25rem;
  content: '';
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c5::before {
  top: 0;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c5::after {
  bottom: 0;
  opacity: 0;
  background: linear-gradient( to bottom, transparent 0%, rgb(177,186,190) 100% );
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c7 {
  margin: 0 auto;
  width: 100%;
  padding: 0.5rem 2rem 2rem;
}

.c8 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 1rem;
}

.c27 {
  max-width: 250px;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1.5rem 1rem 0;
}

.c26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
}

.c26 div:first-child {
  margin-right: 1rem;
  min-width: 5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 100%;
}

.c0 > nav {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > main {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #333333;
  color: #ffffff;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
}

.c12 p > button {
  width: 100%;
}

.c18 {
  margin: 1rem 1rem 1.5rem;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid #808080;
  border-width: 1px 0;
  padding: 1rem 0;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c22 button {
  min-width: 112px;
}

.c22 button[data-size='0'] {
  font-size: 22px !important;
}

.c22 button[data-size='1'] {
  font-size: 28px !important;
}

.c22 button[data-size='2'] {
  font-size: 36px !important;
}

.c21 {
  display: block;
  margin-bottom: 0.4rem;
}

.c15::after {
  display: inline-block;
  margin-left: 0.5rem;
  background: url('/images/arrow-right-open-black.svg') no-repeat;
  width: 1.375rem;
  height: 1.375rem;
  vertical-align: text-bottom;
  content: '';
}

.c17::before {
  display: inline-block;
  margin-right: 0.5rem;
  background: url('/images/arrow-left-open-black.svg') no-repeat;
  width: 1rem;
  height: 1rem;
  vertical-align: text-bottom;
  content: '';
}

@media (min-width:480px) {
  .c9 {
    padding: 1rem 1rem 1rem 4rem;
  }
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

@media (min-width:480px) {
  .c28 {
    line-height: 1.3;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <main
      class="c1"
    >
      <div
        class="c2"
        id="contest-header"
      >
        <div
          class="c3"
          id="audiofocus"
        >
          <h1
            aria-label="United States President and Vice-President."
          >
            <div
              class="c4"
            >
              United States
            </div>
            President and Vice-President
          </h1>
          <p>
            <span
              class=""
            >
              Vote for 
              1
              .
            </span>
             
            <span
              class="screen-reader-only"
            >
              To navigate through the contest choices, use the down button. To move to the next contest, use the right button.
            </span>
          </p>
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <button
                aria-label=" Joseph Barchi and Joseph Hallaren, Federalist."
                class="c9"
                data-choice="barchi-hallaren"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Joseph Barchi and Joseph Hallaren
                    </strong>
                    <br />
                    Federalist
                  </p>
                </div>
              </button>
              <button
                aria-label=" Adam Cramer and Greg Vuocolo, People's."
                class="c9"
                data-choice="cramer-vuocolo"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Adam Cramer and Greg Vuocolo
                    </strong>
                    <br />
                    People's
                  </p>
                </div>
              </button>
              <button
                aria-label=" Daniel Court and Amy Blumhardt, Liberty."
                class="c9"
                data-choice="court-blumhardt"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Daniel Court and Amy Blumhardt
                    </strong>
                    <br />
                    Liberty
                  </p>
                </div>
              </button>
              <button
                aria-label=" Alvin Boone and James Lian, Constitution."
                class="c9"
                data-choice="boone-lian"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Alvin Boone and James Lian
                    </strong>
                    <br />
                    Constitution
                  </p>
                </div>
              </button>
              <button
                aria-label=" Ashley Hildebrand-McDougall and James Garritty, Whig."
                class="c9"
                data-choice="hildebrand-garritty"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Ashley Hildebrand-McDougall and James Garritty
                    </strong>
                    <br />
                    Whig
                  </p>
                </div>
              </button>
              <button
                aria-label=" Martin Patterson and Clay Lariviere, Labor."
                class="c9"
                data-choice="patterson-lariviere"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="c3"
                >
                  <p
                    class="c10"
                  >
                    <strong>
                      Martin Patterson and Clay Lariviere
                    </strong>
                    <br />
                    Labor
                  </p>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
    <nav
      class="c11"
    >
      <div
        class="c12"
      >
        <div
          class="c3"
        >
          <p
            class="c13"
          >
            This is the 
            <strong>
              1st
            </strong>
             of
             
            22
             contests.
          </p>
          <p>
            <button
              class="c14"
              id="next"
              role="option"
              type="button"
            >
              <span
                class="c15"
              >
                Next
              </span>
            </button>
          </p>
          <p>
            <button
              class="c16"
              id="previous"
              role="option"
              type="button"
            >
              <span
                class="c17"
              >
                Back
              </span>
            </button>
          </p>
        </div>
      </div>
      <div
        class="c18"
      >
        <div
          aria-hidden="true"
          class="c19"
        >
          <div
            class="c20"
          >
            <div>
              <div
                class="c21"
              >
                Change Text Size
              </div>
              <div
                class="c22"
              >
                <span
                  class="c23"
                  data-testid="change-text-size-buttons"
                >
                  <button
                    class="c16"
                    data-size="0"
                    type="button"
                    value="0"
                  >
                    A
                  </button>
                  <button
                    class="c24"
                    data-size="1"
                    type="button"
                    value="1"
                  >
                    A
                  </button>
                  <button
                    class="c16"
                    data-size="2"
                    type="button"
                    value="2"
                  >
                    A
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="c25"
          data-testid="election-info"
        >
          <div
            class="c26"
          >
            <div
              aria-hidden="true"
              class="c27"
            />
            <div
              class="c28"
            >
              <h5
                aria-label=" General Election."
              >
                 General Election
              </h5>
              <p
                class="c29"
              >
                November 3, 2020
                <br />
                <span
                  class="c30"
                >
                  Franklin County
                  ,
                </span>
                 
                <span
                  class="c30"
                >
                  State of Hamilton
                </span>
              </p>
              <p
                class="c31"
              >
                 
                
                <span
                  class="c30"
                >
                  <br />
                   Machine ID: 
                  000
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
